src/assets/scripts/client/canvas/CanvasController.js

```diff
@@ -1956,61 +1960,78 @@ export default class CanvasController {
             return;
         }
 
-        const pos = this._toCanvasPosition(aircraft.relativePosition);
-        const rectPos = [0, 0];
-        const rectSize = [CanvasStageModel.width, CanvasStageModel.height];
+        const canvasOrigin = [0, 0];
+        const canvasSize = [CanvasStageModel.width, CanvasStageModel.height];
+        const aircraftPosition = this._toCanvasPosition(aircraft.relativePosition);
+
 
-        // cc.save();
+        cc.save();
+
+        // generic styles of the compass marks
         cc.strokeStyle = this.theme.SCOPE.COMPASS_HASH;
         cc.fillStyle = this.theme.SCOPE.COMPASS_TEXT;
         cc.textAlign = 'center';
         cc.textBaseline = 'middle';
 
-        for (let alpha = 0; alpha < 360; alpha++) {
-            const dir = [
-                sin(degreesToRadians(alpha)),
-                -cos(degreesToRadians(alpha))
-            ];
-
-            const p = positive_intersection_with_rect(pos, dir, rectPos, rectSize);
+        // Compute the intersection point between a ray originating from the
+        // aircraft at a given heading angle and the canvas boundaries, for
+        // each heading angle between 0 and 360 (by 1 degree increment)
+        for (let heading = 0; heading < 360; heading++) {
+            // compute the 2D unit vector representing the ray direction using the given
+            // heading angle
+            const rayUnitVector = vectorize_2d_from_degrees(heading);
+
+            // Use the opposite of the y component of the vector because of the vertical
+            // axes of the reference frames being oriented in opposite directions
+            rayUnitVector[1] = -rayUnitVector[1];
+
+            // compute the intersection point between the ray and the canvas
+            const intersection = positive_intersection_with_rect(
+                aircraftPosition, rayUnitVector, canvasOrigin, canvasSize
+            );
 
-            if (p) {
-                const markLen = (alpha % 5 === 0 ?
-                    (alpha % 10 === 0 ?
-                        16 :
-                        12) :
-                    8
-                );
-                const markWeight = (alpha % 30 === 0 ?
-                    2 :
-                    1
-                );
+            // draw a mark and label at the intersection point
+            // standard marks on all headings
+            // minor marks on headings multiple of 5 degrees
+            // major marks on headings multiple of 10 degrees
+            if (intersection) {
```

Comment:
Reverse the logic here to return early.  That way all the work of this method is not wrapped inside of an if statement